src/components/HeaderButton.tsx

```diff
@@ -1,32 +1,31 @@
 import Icon from "components/Icon";
 import { SortDirection, SortKey } from "hooks/useSort";
 
-export function HeaderButton(
+export function HeaderButton<T>(
   props: React.ButtonHTMLAttributes<HTMLButtonElement> & {
     _sortDirection: SortDirection;
-    _sortKey: SortKey;
-    _activeSortKey: SortKey;
+    _sortKey: SortKey<T>;
+    _activeSortKey: SortKey<T>;
   }
 ) {
   const { _activeSortKey, _sortKey, _sortDirection, children, ...restProps } =
     props;
   return (
     <button
       {...restProps}
-      className="flex items-center justify-start gap-1 uppercase relative"
+      className="flex items-center justify-between gap-1 uppercase"
     >
       <span>{children}</span>
 
       <Icon
-        size={15}
         type={
           _activeSortKey === _sortKey
             ? _sortDirection === "asc"
               ? "Up"
               : "Down"
             : "Unsorted"
         }
-        className={`absolute -right-6 ${
+        className={`w-4 h-4 shrink-0 ${
```

Comment:
Setting the Icon size class name props makes it easier to have responsive sizes when necessary using `tailwind` (e.g. `w-4 md:w-5`)